File: 2149 rearrange-array-elements-by-sign/solution.go
func rearrangeArray(nums []int) []int {
        ans:=make([]int,len(nums))
        l,r:=0,1
        for _,i  := range nums {
            if i < 0 {
                ans[r] = i
                r+=2
            } else {
                ans[l] = i
                l+=2
            }
        }
        return ans
}

File: 2149 rearrange-array-elements-by-sign/solution.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] ans = new int[nums.length];
        int l = 0, r = 1;
        for(int i:nums) {
            if(i < 0) {
                ans[r] = i;
                r+=2;
            } else {
                ans[l] = i;
                l+=2;
            }
        }
        return ans;
    }
}
